migrations/2022-12-05-110642_registration_mode/up.sql

```diff
@@ -0,0 +1,25 @@
+-- create enum for registration modes
+create type registration_mode_enum as enum
+    ('closed', 'require_application', 'open');
```

Comment:
I'm wondering if these need to be exclusive options. IE in the same way that verify_email is optional, require_application is currently optional.

This makes it so that additional signup verifications methods must be exclusive, IE if we do the content approvals thing, then making them exclusive means you can't turn on both.

Reply:
I dont think it makes much sense to enable both at the same time. And more importantly, I think that this kind of enum for registration modes is much easier to understand. It means just going through the possible options and picking the one thats most appropriate. If we keep adding bools, it becomes very confusing which settings are related to registration, and how exactly they they will interact in different combinations.

Reply:
Did you want to get this in before 0.17.0?

Reply:
Yes because its a breaking change. Then we can add the new registration mode in 0.17.x without breaking backwards compat.